tools/biosnoop.bt

```diff
@@ -1,43 +1,64 @@
 #!/usr/bin/env bpftrace
 /*
  * biosnoop.bt   Block I/O tracing tool, showing per I/O latency.
- *               For Linux, uses bpftrace, eBPF.
+ *	       For Linux, uses bpftrace, eBPF.
  *
- * TODO: switch to block tracepoints. Add device, offset, and size columns.
+ * TODO: switch to block tracepoints.
  *
  * This is a bpftrace version of the bcc tool of the same name.
  *
  * 15-Nov-2017	Brendan Gregg	Created this.
+ * 31-Jul-2020  Trent Lloyd     Added support for disk name, offset and length.
+ *       			Output systm time at startup. Requires bpftrace
+ *      			> 0.11.0 for more than 6 printf arguments.
  */
 
+#include <linux/blkdev.h>
+
 BEGIN
 {
-	printf("%-12s %-16s %-6s %7s\n", "TIME(ms)", "COMM", "PID", "LAT(ms)");
+	time("Start Time: %F %T\n");
```

Comment:
I would not add this start time. It's clutter, and there's the risk people will add it (and more) to all tools. This is the kind of thing I put behind an option flag in the bcc version; we don't have option flags in bpftrace yet, although even if we did, I'm not sure we want the bpftrace tools to grow this much.

I sometimes automate these tools and want details such as the date and other details stored with the output: so I simply run commands like "date; uname -a; uptime" before the tool.

Reply:
The problem I had here is I sometimes see multi-second delays to the BPF startup that makes that method quite innacurate for correlating against other logs.

I'll remove this but perhaps we could solve that by adding a generic "print timestamp at startup" option to bpftrace itself that correctly correlates with the elapsed counter.